ngraph/test/util/unary_test.hpp

```diff
@@ -0,0 +1,293 @@
+                    value[i] = d(gen);
+            }
+
+            // random (uniform-real-distrubition) generate data
+            Data(rand_uniform_real_tag, double low, double high, const ngraph::Shape& s)
+                : value(ngraph::shape_size(s))
+                , shape(s)
+            {
+                std::mt19937 gen{0};
+                std::uniform_real_distribution<> d{low, high};
+                for (size_t i = 0; i < value.size(); ++i)
+                    value[i] = d(gen);
+            }
+
+            Data() = default;
+
+            // only vector, no shape specified
+            Data(const std::vector<T>& v)
+                : value(v)
+                , shape{}
+                , no_shape(true)
+            {
+            }
+
+            // tensor with shape
+            Data(const std::vector<T>& v, const ngraph::Shape& s)
+                : value(ngraph::shape_size(s))
+                , shape{s}
+            {
+                for (size_t i = 0; i < value.size(); ++i)
+                    value[i] = v[i % v.size()];
+            }
+
+            // caller have to pass two level of braces when aggregate-initialize input argument of
+            // type std::vector<T>. by adding an overload of type std::initializer_list<T>, we can
+            // save one level of braces
+            Data(const std::initializer_list<T>& v)
+                : value(v.size())
+                , shape{}
+                , no_shape(true)
+            {
+                std::copy(v.begin(), v.end(), value.begin());
+            }
+
+            // we would duplicate by circular-copy or truncate initializer list when number of
+            // element provided in initializer list is not matching the desired shape
+            Data(const std::initializer_list<T>& v, const ngraph::Shape& s)
+                : value(ngraph::shape_size(s))
+                , shape{s}
+            {
+                auto it = v.begin();
+                for (size_t i = 0; i < value.size(); ++i, ++it)
+                {
+                    if (it == v.end())
+                        it = v.begin();
+                    value[i] = *it;
+                }
+            }
+
+        private:
+            std::vector<T> value{};
+            ngraph::Shape shape{};
+            bool no_shape = false;
+
+            template <typename TestEngine, ngraph::element::Type_t et>
+            friend class unary_test;
+        };
+
+        template <typename TestEngine, ngraph::element::Type_t et>
+        class unary_test
+        {
+        public:
+            unary_test(std::shared_ptr<ngraph::Function> function)
+                : m_function(function)
+            {
+            }
+
+            template <ngraph::element::Type_t eleType = et>
+            void test(const std::initializer_list<Data<eleType>>& inputs,
+                      const std::initializer_list<Data<eleType>>& expeted,
+                      Tolerance tol = {})
```

Comment:
IMHO, now test is overloading horror.

Reply:
Yes, but now we can cover more test case with multiple inputs/outputs. I think it's affordable.